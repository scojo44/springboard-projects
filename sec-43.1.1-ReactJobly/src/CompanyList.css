.CompanyList {
  display: flex;
  flex-direction: column;
  width: 80%;
  margin: 0 auto;
}
.CompanyList ul {
  flex: 1;
  margin: 0;
  padding: 0;
  overflow-y: scroll;
}
