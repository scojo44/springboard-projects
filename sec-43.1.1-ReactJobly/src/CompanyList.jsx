import React, {useEffect, useState} from 'react'
import JoblyApi from './api'
import Alert from './Alert'
import SearchForm from './SearchForm'
import CompanyCard from './CompanyCard'
import './CompanyList.css'

export default function CompanyList() {
  const [companies, setCompanies] = useState([]);
  const [query, setQuery] = useState();
  const [error, setError] = useState('');

  useEffect(() => {
    async function fetchCompanies() {
      try {
        const firms = await JoblyApi.getCompanies(query);
        setCompanies(() => [...firms]);
        setError('');
      }
      catch(e) {
        setError(e);
      }
    }

    fetchCompanies();
  }, [query]);

  return (
    <section className="CompanyList">
      {error && <Alert type="error" messages={[`Error loading companies: ${error}`]} />}
      <SearchForm fieldName="nameLike" returnQuery={search => setQuery(() => search)} />
      <ul>
        {companies.map(c => <CompanyCard company={c} key={c.handle} />)}
      </ul>
    </section>
  );
}
