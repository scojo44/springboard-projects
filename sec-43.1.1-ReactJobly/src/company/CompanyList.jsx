import React from 'react'
import JoblyApi from '../api'
import useJoblyAPI from '../hooks/useJoblyAPI'
import LoadingIndicator from '../widgets/LoadingIndicator'
import Alert from '../widgets/Alert'
import SearchForm from '../widgets/SearchForm'
import CompanyCard from './CompanyCard'
import './CompanyList.css'

export default function CompanyList() {
  const {data: companies, setParams, error, isLoading} = useJoblyAPI(JoblyApi.getCompanies);

  if(isLoading) return <LoadingIndicator />;

  return (
    <section className="CompanyList">
      <SearchForm fieldName="nameLike" returnQuery={query => setParams(query)} />
      {error
        ? <Alert type="error" messages={[`Error loading companies: ${error}`]} />
        : <ul>
            {companies.map(c => <CompanyCard company={c} key={c.handle} />)}
          </ul>
      }
    </section>
  );
}
