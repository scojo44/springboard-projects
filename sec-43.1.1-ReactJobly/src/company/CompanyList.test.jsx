import {render, waitForElementToBeRemoved} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import JoblyApi from '../api'
import CompanyList from './CompanyList'

const TEST_COMPANIES = [{
  handle: 'test-corp',
  name: 'Test Corp.',
  description: 'Test Corp is the leading.... yada yada yada',
  numEmployees: 5,
  logoURL: '/test-corp-logo.png',
  jobs: []
},
{
  handle: 'test-llc',
  name: 'Test LLC',
  description: 'Test LLC is the leading.... blah blah blah',
  numEmployees: 10,
  logoURL: '/test-llc-logo.png',
  jobs: []
}]

// Mock the API hook
JoblyApi.getCompanies = vi.fn(() => TEST_COMPANIES);

describe('CompanyList Tests', () => {
  it('Renders without crashing', async () => {
    const {getByText} = render(
      <MemoryRouter>
        <CompanyList />
      </MemoryRouter>
    );

    await waitForElementToBeRemoved(getByText('Loading...'));
  });

  it('Matches snapshot', async () => {
    const {asFragment, getByText} = render(
      <MemoryRouter>
        <CompanyList />
      </MemoryRouter>
    );

    await waitForElementToBeRemoved(getByText('Loading...'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('Lists some companies', async () => {
    const {getByText} = render(
      <MemoryRouter>
        <CompanyList />
      </MemoryRouter>
    );

    await waitForElementToBeRemoved(getByText('Loading...'));
    expect(getByText(TEST_COMPANIES[0].name)).toBeInTheDocument();
    expect(getByText(TEST_COMPANIES[0].description)).toBeInTheDocument();
    expect(getByText(TEST_COMPANIES[1].name)).toBeInTheDocument();
    expect(getByText(TEST_COMPANIES[1].description)).toBeInTheDocument();
  });
});
