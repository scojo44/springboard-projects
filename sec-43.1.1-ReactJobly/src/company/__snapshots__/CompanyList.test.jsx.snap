// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompanyList Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="CompanyList"
  >
    <form
      class="SearchForm"
    >
      <input
        id="nameLike"
        name="nameLike"
        placeholder="Enter search term"
        type="text"
        value=""
      />
      <button
        type="button"
      >
        Search
      </button>
    </form>
    <ul>
      <li
        class="CompanyCard"
      >
        <a
          class="link"
          href="/test-corp"
        >
          <div>
            <h3>
              Test Corp.
            </h3>
            <p>
              Test Corp is the leading.... yada yada yada
            </p>
          </div>
          <img
            alt="Test Corp. Logo"
            src="/test-corp-logo.png"
          />
        </a>
      </li>
      <li
        class="CompanyCard"
      >
        <a
          class="link"
          href="/test-llc"
        >
          <div>
            <h3>
              Test LLC
            </h3>
            <p>
              Test LLC is the leading.... blah blah blah
            </p>
          </div>
          <img
            alt="Test LLC Logo"
            src="/test-llc-logo.png"
          />
        </a>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
